c/meson.build

```diff
@@ -23,12 +23,15 @@ project(
     meson_version: '>=1.3.0',
     default_options: [
         'buildtype=release',
-        'c_std=gnu99',  # TODO: gmtime_r is gnu99 - likely unintentional std
+        'c_std=c99',
         'warning_level=2',
         'cpp_std=c++17',
     ]
 )
 
+add_project_arguments('-Wno-int-conversion', '-Wno-unused-parameter', language: 'c')
```

Comment:
Not required for this PR but these are all the warnings showing up in Meson currently. Could tackle these as follow ups to get us to a clean build